examples/20_tokio_tradeclient/src/main.rs

```diff
@@ -37,15 +37,15 @@ fn main() -> anyhow::Result<()> {
     Ok(())
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, strum::EnumVariantNames, strum::EnumString)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, strum::EnumIter, strum::Display)]
 enum UserAction {
     EnterOrder,
     Quit,
 }
 
 fn prompt_user_action() -> anyhow::Result<UserAction> {
-    let s = inquire::Select::new("Select an action", UserAction::VARIANTS.to_vec()).prompt()?;
-    Ok(UserAction::from_str(&s)?)
+    let s = inquire::Select::new("Select an action", UserAction::iter().collect()).prompt()?;
```

Comment:
While I have your attention, based on your very simple usage (no further customization on the prompt), how much would you prefer the current syntax (using the full builder pattern) vs this function call?

```suggestion
    let s = inquire::prompt_selection("Select an action", UserAction::iter().collect())?;
    
    // or perhaps we could also omit inquire here since the fn name is verbose enough
    let s = prompt_selection("Select an action", UserAction::iter().collect())?;
```

Reply:
I like this a lot, thank you for the suggestion! I'll apply it.

Reply:
Mmm, I'm actually having a hard time finding the `prompt_selection` API you're suggesting. Neither a search on docs.rs nor one on the `inquire` GH repo give any results. I might be missing something here 🤔

Reply:
I was misleading sorry... this function hasn't been released yet, but I'm toying around with the idea of exposing a "one-shot" api for the simplest of use cases. Either something like `prompt_confirmation(msg)`, or maybe a macro where I can still have a small degree of customization `prompt_confirmation!($msg) {} prompt_confirmation!($msg, $default) {}` you know?

Reply:
Gotcha! Yeah I'd use `prompt_selection`, but only if a similar option is available for `Text` – I'd prioritize consistency over saving a few keystrokes. As for macros, I tend to avoid them unless absolutely necessary.

Reply:
>As for macros, I tend to avoid them unless absolutely necessary.

It's good to hear that upfront. May I ask why? I don't know a lot about its impacts, or if they are a zero-cost abstraction at all.

Reply:
They are zero-cost indeed, and extremely useful as you can do lots of things you normally wouldn't be able to do. I'm using quite a few in FerrumFIX alone so I'm not against them in principle.

There are two major drawbacks, the way I see it, and they are quite related with each other. (1) The error messages you get with them are quite cryptic and (2) they degrade IDE development experience, as you lose autocomplete, type hints, and other goodies. Also, debugging proc macro -generated code can be very painful, although that almost only concerns the crate author and not end users. For these reasons, I tend to limit proc macro usage to situations where alternatives are way too verbose or outright don't work. Declarative macros somewhat share these problems but they tend to be nicer to work with, in my experience.

Reply:
Oh, I forgot: they also quite increase compilation times.